bitcoin/src/blockdata/transaction.rs

```diff
@@ -526,6 +534,20 @@ impl TxOut {
         Weight::from_vb(self.size() as u64).expect("should never happen under normal conditions")
     }
 
+    /// Returns the effective_value `Amount` returning `None` on overflow.
+    ///
+    /// The effective value is the value of this output minus the amount to spend it.  That is,
+    /// the effective_value can be calculated as value - (fee_rate * weight).
+    pub fn effective_value(
+        &self,
+        fee_rate: FeeRate,
+        satisfaction_weight: Weight,
```

Comment:
Needs to be documented. Also we should have constants for standard scripts like P2TR.

Reply:
> Also we should have constants for standard scripts like P2TR.

Yes, that would be nice, but I may punt on that until later since I'd like this PR small so it can be reviewed and merged quicker and easier.  Seems like a good follow commit though.

Reply:
OK, I think extra feature can go into later PR.

Reply:
done, marking resolved.